src/lro/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types and functions to make LROs easier to use and to require less boilerplate.
+
+use gax::error::Error;
+use gax::Result;
+use std::future::Future;
+use std::marker::PhantomData;
+
+/// The result of polling a Long-Running Operation (LRO).
+///
+/// # Parameters
+/// * `R` - the response type. This is the type returned when the LRO completes
+///   successfully.
+/// * `M` - the metadata type. While operations are in progress the LRO may
+///   return values of this type.
+#[derive(Debug)]
+pub enum PollingResult<R, M> {
+    /// The operation is still in progress.
+    InProgress(Option<M>),
+    /// The operation completed. This includes the result.
+    Completed(Result<R>),
+    /// An error trying to poll the LRO.
+    ///
+    /// Not all errors indicate that the operation failed. For example, this
+    /// may fail because it was not possible to connect to Google Cloud. Such
+    /// transient errors may disappear in the next polling attempt.
+    ///
+    /// Other errors will never recover. For example, a [ServiceError] with
+    /// a [NOT_FOUND], [ABORTED], or [PERMISSION_DENIED] code will never
+    /// recover.
+    ///
+    /// [ServiceError]: gax::error::ServiceError
+    /// [NOT_FOUND]: rpc::model::code::NOT_FOUND
+    /// [ABORTED]: rpc::model::code::ABORTED
+    /// [PERMISSION_DENIED]: rpc::model::code::PERMISSION_DENIED
+    PollingError(Error),
+}
+
+/// The result of starting and polling a long-running operation.
+///
+/// In most cases this is backed by [Operation][longrunning::model::Operation].
+pub trait Operation {
+    type ResponseType;
+    type MetadataType;
+
+    fn name(&self) -> String;
+    fn done(&self) -> bool;
+    fn metadata(&self) -> Option<&wkt::Any>;
+    fn response(&self) -> Option<&wkt::Any>;
+    fn error(&self) -> Option<&rpc::model::Status>;
+}
+
+/// Implements [Operation] using [longrunning::model::Operation].
+pub struct TypedOperation<R, M> {
```

Comment:
nit: could this be called Operation, and the trait be renamed to something like `LongrunningOperation`?